src/bulk/common.ts

```diff
@@ -684,10 +684,13 @@ export class MongoBulkWriteError extends MongoServerError {
     super(error as Error);
     Object.assign(this, error);
 
-    this.name = 'MongoBulkWriteError';
     this.result = result;
   }
 
+  get name(): string {
```

Comment:
I would advise against using getters just for the sake of making something readonly - adding a function to a class does make it heavier (less performant) than just having a plain property; typically getters are used when the corresponding attribute needs to be computed dynamically; if it is important to make this particular property readonly, `defineProperty` allows designating it as such

Reply:
> adding a function to a class does make it heavier (less performant) than just having a plain property

You can generally expect that this kind of getter is overall faster than a plain property, because it lives on the prototype and does not require the creation of an additional property on each object instance.

Reply:
That is an interesting point; my original comment was coming from the perspective of accessing the property itself via getter vs direct read, not the creation of the error object itself; having properties defined on the prototype instead of the instance would indeed make for a smaller memory footprint for the individual instances (not that we likely have anything to worry about in practice since it's unlikely anyone is compiling or iterating over large arrays of our error objects)

Reply:
we could consider setting a static property on the class that has the error name and then have `this.name = <static prop>`, which might get us the best of both worlds; @addaleax what do you think?

Reply:
Practically speaking, the cost of accessing a trivial getter is zero because engines can inline these functions with the same efficiency as an access to a plain data property.

If you want to emulate what e.g. the built-in JS Error types do, you can define a readonly data property on the prototype; but practically speaking, I think using a getter for `Error` subclasses is a common pattern and it doesn’t really hurt to do it.

Reply:
I have the opposing stylistic preference here but I'm happy to defer to whichever option the rest of the team feels most comfortable with. Thank you for the perspective :)